.github/workflows/python-poetry-ci.yml

```diff
@@ -33,6 +45,9 @@ jobs:
 
       - name: Setup Poetry
         uses: matrix-org/setup-python-poetry@v1
+        with:
+          # We want to make use of type hints in optional dependencies too.
+          extras: "${{ inputs.typechecking-extras }}"
```

Comment:
Seems OK -- should we use just `extras` for this though instead of `typechecking-extras`?

Reply:
I wanted to emphasise that this was only going to install extras to run mypy; flake8, isort, black all run from the "bare" set of developer dependencies that poetry defines.

Not a strongly-held opinion though; happy to change if you think `extras` is clearer.

Reply:
👍 Ah, I see. I think it is fine either way, just wasn't sure of the rationale!